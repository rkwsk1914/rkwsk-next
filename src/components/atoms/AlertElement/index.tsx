import * as React from 'react'

import Alert from '@mui/material/Alert'
import { AlertColor } from '@mui/material/Alert'
import AlertTitle from '@mui/material/AlertTitle'

type Props = {
  children?: React.ReactNode
  onClose?: (event: React.SyntheticEvent) => void;
  type: AlertColor
  variant?: 'standard' | 'filled' | 'outlined'
};

export const AlertElement: React.FC<Props> = (
  {
    children,
    onClose,
    type = 'success',
    variant = 'filled'
  }
): JSX.Element => {
  const setTitle = type.charAt(0).toUpperCase() + type.slice(1)

  return (
    <Alert
      severity={type}
      variant={variant}
      onClose={onClose}
    >
      <AlertTitle>{setTitle}</AlertTitle>
      {children}
    </Alert>
  )
}
