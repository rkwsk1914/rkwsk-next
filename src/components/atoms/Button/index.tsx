import * as React from 'react'

import clsx from 'clsx'
import Link from 'next/link'
import { Link as ScrollLink } from 'react-scroll'

import { useGetDarkModeStyleClass } from '@/hooks/useGetDarkModeStyleClass'

import { ICON_DATA } from '@/const/IconData'


import styles from './style.module.scss'


interface BaseProps {
  text: string
  type?: 'prime' | 'dangerous' | 'warning' | 'standard'
  onClick?: () => void
}

interface LinkProps extends BaseProps {
  href?: string
  toId?: never
}

interface ScrollProps extends BaseProps {
  href?: never
  toId?: string
}

type Props = LinkProps | ScrollProps;

export const Button: React.FC<Props> = (
  {
    text,
    type,
    href,
    toId,
    onClick
  }
): JSX.Element => {
  const darkClassName = useGetDarkModeStyleClass(styles.button, styles.dark)
  const buttonClassName = clsx(darkClassName, {
    [styles.prime]: type === 'prime',
    [styles.dangerous]: type === 'dangerous',
    [styles.warning]: type === 'warning',
    [styles.standard]: type === 'standard',
  })

  if (href) return (
    <Link className={buttonClassName} href={href}>
      {text}<span className={styles.arrow}>{ICON_DATA.arrowRight}</span>
    </Link>
  )

  if (toId) return (
    <ScrollLink className={buttonClassName} to={toId} smooth onClick={onClick}>{text}</ScrollLink>
  )

  return (
    <button className={buttonClassName} onClick={onClick}>{text}</button>
  )
}
