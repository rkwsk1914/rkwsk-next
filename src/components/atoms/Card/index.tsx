import * as React from 'react'

import { useGetDarkModeStyleClass } from '@/hooks/useGetDarkModeStyleClass'

import styles from './style.module.scss'

type RefProps = {
  title?: string
  children?: React.ReactNode
  matchHeight?: number | null
};

export const Card = React.forwardRef(
  function RefComponent (
    {
      title,
      children,
      matchHeight
    }: RefProps,
    ref?: React.Ref<HTMLDivElement> | null
  ): JSX.Element {
    const cardClassName = useGetDarkModeStyleClass(styles.card, styles.dark)
    const cardHeight = matchHeight ? `${matchHeight}px` : 'auto'

    return (
      <div className={cardClassName} ref={ref} style={{ minHeight: cardHeight }}>
        {title && <div className={styles.head}>{title}</div>}
        <div className={styles.body}>
          {children}
        </div>
      </div>
    )
  }
)
