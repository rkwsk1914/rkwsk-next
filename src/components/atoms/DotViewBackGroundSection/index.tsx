import * as React from 'react'

import Image from 'next/image'

import styles from './style.module.scss'


type Props = {
  children?: React.ReactNode
  imgSrc: string
};

export const DotViewBackGroundSection: React.FC<Props> = ({
  children,
  imgSrc
}): JSX.Element => {
  return (
    <section className={styles.wrap}>
      <div className={styles.content}>{children}</div>
      <div className={styles.bg_dot}></div>
      <Image
        src={imgSrc}
        alt='background-image'
        width={500}
        height={500}
        className={styles.image} />
    </section>
  )
}
