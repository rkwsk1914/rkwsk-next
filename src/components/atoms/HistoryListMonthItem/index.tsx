import * as React from 'react'

import { useGetDarkModeStyleClass } from '@/hooks/useGetDarkModeStyleClass'
import { MonthType } from '@/types/MonthType'

import styles from './style.module.scss'

type Props = {
  month?: MonthType
  children?: React.ReactNode
};

export const HistoryListMonthItem: React.FC<Props> = (
  {
    children,
    month
  }
): JSX.Element => {
  const boxClassName = useGetDarkModeStyleClass(styles.box, styles.dark)
  return (
    <div className={boxClassName}>
      {month && <span className={styles.month}>{month}æœˆ</span>}
      <p className={styles.content}>{children}</p>
    </div>
  )
}
