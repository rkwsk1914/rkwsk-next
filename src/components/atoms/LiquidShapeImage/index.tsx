import * as React from 'react'

import Image from 'next/image'

import { useGetDarkModeStyleClass } from '@/hooks/useGetDarkModeStyleClass'

import styles from './style.module.scss'

type Props = {
  image: Omit<React.ComponentProps<typeof Image>, 'className' | 'alt'>
  alt: string
};

export const LiquidShapeImage: React.FC<Props> = (
  {
    image,
    alt
  }
): JSX.Element => {
  const classNameContainer = useGetDarkModeStyleClass(styles.container, styles.dark)
  return (
    <div className={classNameContainer}>
      <div className={styles.liquid} />
      <div className={styles.liquid2} />
      <Image className={styles.image} alt={alt} {...image} />
    </div>
  )
}
