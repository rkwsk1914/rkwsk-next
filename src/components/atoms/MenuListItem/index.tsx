import * as React from 'react'

import Link from 'next/link'

import { useGetDarkModeStyleClass } from '@/hooks/useGetDarkModeStyleClass'
import { MenuDataType } from '@/types/MenuDataType'

import styles from './style.module.scss'


type Props = {
  data: MenuDataType
};

const Content: React.FC<Props> = ({ data }): JSX.Element => {
  const className = useGetDarkModeStyleClass(styles.li, styles.dark)
  return (
    <li className={className}>
      <Link href={data.href}>{data.text}</Link>
    </li>
  )
}

export const MenuListItem =  React.memo(Content)
