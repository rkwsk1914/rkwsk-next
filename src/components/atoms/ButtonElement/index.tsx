import * as React from 'react';

import Button from '@mui/material/Button';

type Props = {
  children?: React.ReactNode
  color?: 'inherit' | 'primary' | 'secondary' | 'success' | 'error' | 'info' | 'warning'
  variant?: 'contained' | 'outlined' | 'text'
  size?: 'small' | 'medium' | 'large'
  disabled?: boolean
};

export const ButtonElement: React.FC<Props> = (
  {
    children,
    color,
    variant,
    size,
    disabled
  }
): JSX.Element => {
  return (
    <Button
      color={color}
      variant={variant}
      size={size}
      disabled={disabled}
    >
      {children}
    </Button>
  );
}
