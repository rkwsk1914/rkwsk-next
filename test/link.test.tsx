import React from 'react'

import { render } from '@testing-library/react'

import { isScrollLink } from '@/components/atoms/MenuListItem'

describe('scroll Link href test', () => {

  const donTest = (testHref: string, toBe: boolean) => {
    test(testHref, () => {
      expect(isScrollLink(testHref)).toBe(toBe)
    })
  }

  donTest('#menu', true)
  donTest('#menu-1', true)
  donTest('#1-menu', true)
  donTest('menu', false)
  donTest('/menu', false)
  donTest('https://react.dev/', false)
  donTest('https://react.dev/learn/writing-markup-with-jsx#jsx-putting-markup-into-javascript', false)
})

