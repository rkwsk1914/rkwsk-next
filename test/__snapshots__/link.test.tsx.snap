// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<li
  className="li"
>
  <a
    onClick={[Function]}
    style={{}}
  >
    menu1
  </a>
</li>
`;
